va/validation-authority_test.go

```diff
@@ -213,12 +218,19 @@ func TestSimpleHttpTLS(t *testing.T) {
 	finChall, err := va.validateSimpleHTTP(ident, chall)
 	test.AssertEquals(t, finChall.Status, core.StatusValid)
 	test.AssertNotError(t, err, chall.Path)
+	logs := log.GetAllMatching(`^\[AUDIT\] Attempting to validate SimpleHTTPS for `)
+	test.AssertEquals(t, len(logs), 1)
+	test.AssertEquals(t, logs[0].Priority, syslog.LOG_NOTICE)
 }
 
 func TestSimpleHttp(t *testing.T) {
 	va := NewValidationAuthorityImpl(true)
 	va.DNSResolver = &mocks.MockDNS{}
 
+	log := mocks.NewSyslogWriter()
+	defer log.Close()
+	defer mocks.SwitchLog(mocks.SwitchLog(log))
+
```

Comment:
Yes, that's the intent here.

Even though it wouldn't affect the test outcomes, I'm a bit bothered by the prospect of having logs for some test cases (the ones that don't test their log output) unpredictably going to the real syslog or the mock, depending on which _other_ test cases are being run and in what order. (I think this would happen if we called `UseMockLog()` at the top of a test case.)

The "log to a mock when testing component X" (i.e., `UseMockLog()` in global scope) and "log to a mock during test function Y, but restore when done" approaches were the two ways I came up with to make logging during test runs (not just the test results) predictable. And I'm more inclined to go with the first option -- and do it for all components -- primarily because it seems better to me to have the test suite output say "would have syslogged this" than to generate real syslog entries on the dev/test machine.

Your call, though.